pkg/logs/internal/tailers/file/tailer.go

```diff
@@ -277,12 +290,20 @@ func (t *Tailer) readForever() {
 		log.Info("Closed", t.file.Path, "for tailer key", t.file.GetScanKey(), "read", t.Source().BytesRead.Get(), "bytes and", t.decoder.GetLineCount(), "lines")
 	}()
 
+	getCurrentTime := func() int64 {
+		return time.Now().Unix()
+	}
+	ms := NewMovingSum(86400, getCurrentTime)
+
 	for {
 		n, err := t.read()
 		if err != nil {
 			return
 		}
 		t.recordBytes(int64(n))
+		ms.AddDataPoint(int64(n))
+		sum := ms.CalculateMovingSum()
```

Comment:
Similar to `StatsTracker`, I think your new struct should implement the `InfoProvider` interface, so the sum can be calculated and queried on-demand. You would register it with the tailer info. 

There is a small problem with computing the sum on-read:

Imagine a file is very busy for 1 hour, and then totally inactive for 2 days. 
The sum will have not been updated for 2 days
So when we request agent status, we will get an incorrect measurement. The sum will only become correct again if more data is written to the file.